src/orquestra/sdk/_base/_storage/_basemodel.py

```diff
@@ -0,0 +1,29 @@
+################################################################################
+# © Copyright 2024 Zapata Computing Inc.
+################################################################################
+
+from typing import Any
+
+from pydantic.main import BaseModel
+
+
+class OrquestraBaseModel(BaseModel):
+    """The pydantic BaseModel changed between V1 and V2.
+
+    As a result, workflow outputs generated prior to the V2 upgrade may not be
+    depickled correctly. The culpret is a change in behaviour of `__setstate__`.
+
+    This class adds a new `__setstate__` that wraps the V2 BaseModel `__setstate__` and
+    adds the missing behaviour back in.
+
+
+    """
```

Comment:
**praise**: wow, what a great hack! The issue with depickling old objects sounded really bad; I'm glad you found a solution!

Reply:
I can't claim all the credit, the internet helped a lot!